assets/model_monitoring/components/src/model_data_collector_preprocessor/store_url.py

```diff
@@ -57,11 +57,14 @@ def _get_url(self, scheme=None, store_type=None, relative_path=None) -> str:
 
     def get_credential(self) -> Union[str, ClientSecretCredential, AzureSasCredential, None]:
         """Get credential for this store url."""
-        def get_default_credential():
-            return DefaultAzureCredential() if self._is_secure() else None
-
         if not self._datastore:
-            return get_default_credential()
+            if self._is_secure():
+                raise InvalidInputError(
+                    "abfss, wasbs and https URLs are credential less and thus NOT supported for Model Monitoring job, "
+                    "please use credential data instead.")
```

Comment:
Is there any instructions we can give to user for how to add credential to data or is the only option to choose a different data asset with credential?

Reply:
For now, we only support credential data. In the future, we need to discuss with PM whether and how we want to support credential less data.